compiler/qsc_eval/src/lower.rs

```diff
@@ -81,6 +81,8 @@ impl Lowerer {
         fir_package: &mut fir::Package,
         hir_package: &hir::Package,
     ) -> Vec<StmtId> {
+        let new_entry = hir_package.entry.as_ref().map(|e| self.lower_expr(e));
```

Comment:
Since @cesarzc is taking a dependency on FIR for RCA and has refactored lowering some, it might be worth verifying with him that this change to how these expressions are tracked won't invalidate any assumptions or logic there.

Reply:
This change does not invalidate assumptions in RCA. The only adjustment I have to make is to include an entry expression in the analysis data structure to save the properties of the entry expression.

Reply:
On a related note, @minestarks does it matter the order in which things are lowered? I would expect to first lower items, then top-level statements and then the entry expression since it can depend on items and top-level statements.

Reply:
Good question. I don't think it matters in practice, because I can see some existing unit tests that would have been broken if it mattered :) But to keep the order more logical, I'll follow your suggestion and do the entry expression last.